### PR TITLE
Propertdd

### PR DESCRIPTION
**Pull request by nickbdyer**
Lots of commits for very small steps tdd.

### CODE REVIEW

File: src/word_wrap/WrapperTest.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public void emptyString() {
 
     @Test
     public void wordShorterThanWrapColumn() {
-        assertEquals("word", Wrapper.wrap("word", 7));
```
Comment (nickbdyer) [inline src/word_wrap/WrapperTest.java]: These test changes felt very hacky, but also meant that the next refactor was much simpler to see.

File: src/word_wrap/Wrapper.java [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
 
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
-
 public class Wrapper {
 
     public static String wrap(String s, int column) {
         if (s.length() < column) {
             return s;
         }
-        if (column > 1) {
```
Comment (nickbdyer) [inline src/word_wrap/Wrapper.java]: Is this adding functionality?

File: src/word_wrap/Wrapper.java [hunk 3]
```diff
@@ -3,9 +3,9 @@
 public class Wrapper {
 
     public static String wrap(String s, int column) {
-        if (s.length() < column) {
+        if (s.length() <= column) {
             return s;
         }
-        return s.substring(0, column) + "\n" + s.substring(column, s.length());
+        return s.substring(0, column) + "\n" + wrap(s.substring(column, s.length()), column);
```
Comment (nickbdyer) [inline src/word_wrap/Wrapper.java]: Too big a step? Is iteration considered a better step here?

Comment (mateuadsuara) [inline src/word_wrap/Wrapper.java]: No, recursion seems to be a better smaller step than iteration.
Based on Uncle Bob's Transformation Priority Premise (https://blog.8thlight.com/uncle-bob/2013/05/27/TheTransformationPriorityPremise.html), `statement->recursion` is given precedence over `if->while`

Comment (nickbdyer) [inline src/word_wrap/Wrapper.java]: Just read it. And your blog posts. Great reads. So funny that Uncle Bob's Post is all about Word Wrap too!

**Pull request merged**